SmartAppointmentSystem.Api/Controllers/AppointmentController.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SmartAppointmentSystem.Api.Extensions;
+using SmartAppointmentSystem.Api.Models;
+using SmartAppointmentSystem.Business.Contracts;
+
+namespace SmartAppointmentSystem.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class AppointmentController(IAppointmentService appointmentService) : ControllerBase
+{
+    [HttpPost]
+    public async Task<IActionResult> CreateAppointment([FromBody] AppointmentRequestModel requestModel)
+    {
+        var fill = requestModel.Map();
+        var gettingFilled = await appointmentService.CreateAppointmentAsync(fill);
+        if (!gettingFilled)
+        {
+            return StatusCode(500, "Appointment could not create");
+        }
+        return Ok(gettingFilled);
+    }
+    [HttpGet("all")]
+    public async Task<IActionResult> GetAllAppointments()
+    {
+        var getAllApp = await appointmentService.GetAllAppointmentsAsync();
+        if (getAllApp.Count < 1 || getAllApp == null)
+        {
+            return NotFound("There is no appointment.");
+        }
+        return Ok(getAllApp);
+    }
+
+    [HttpGet("{id}")]
+    public async Task<IActionResult> GetAppointment(Guid id)
+    {
+        var getAppo = await appointmentService.GetAppointmentsByIdAsync(id);
+        if (getAppo == null)
+        {
+            return NotFound();
+        }
+        return Ok(getAppo);
+    }
+
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
+    [HttpGet("patient-appointments")]
+    public async Task<IActionResult> GetPatientsAppointments()
+    {
+        var userId = HttpContext.User.GetUserId();
+        var appointments = await appointmentService.GetUserAppointmentsAsync(userId);
+
+        if (appointments == null || !appointments.Any())
+        {
+            return NotFound("There is no appointment.");
```

Comment:
Sometimes you return response message for NotFound, and sometimes you do not. I guess you don't need a message for NotFound since its status code explicitly defines the situation. So just `return NotFound();`

Do this for all the places you return NotFound with a response message.